axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveViewHelperServiceImpl.java

```diff
@@ -29,19 +29,22 @@ public class MoveViewHelperServiceImpl implements MoveViewHelperService {
 
   @Override
   public String filterPartner(Company company, Journal journal) {
-    Long companyId = company.getId();
-    String domain = "self.isContact = false AND " + companyId + " member of self.companySet";
-    if (journal != null && !Strings.isNullOrEmpty(journal.getCompatiblePartnerTypeSelect())) {
-      domain += " AND (";
-      String[] partnerSet = journal.getCompatiblePartnerTypeSelect().split(", ");
-      String lastPartner = partnerSet[partnerSet.length - 1];
-      for (String partner : partnerSet) {
-        domain += "self." + partner + " = true";
-        if (!partner.equals(lastPartner)) {
-          domain += " OR ";
+    String domain = "self.isContact = false";
+    if (company != null) {
+      Long companyId = company.getId();
+      domain = " AND " + companyId + " member of self.companySet";
```

Comment:
There is a missing + that triggers an error when company is empty 
Also, maybe it's not necessary to store companyId in a variable, we can use it directly in query
```suggestion
      domain += " AND " + company.getId() + " member of self.companySet";
```